Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/role_icon.tsx

```diff
@@ -19,50 +19,52 @@ interface State {
 }
 
 /** Displays a panel of icons highlighting an account's roles. */
-export class RolesIcon extends React.Component<Properties, State> {
+export class RoleIcon extends React.Component<Properties, State> {
   constructor(props: Properties) {
     super(props);
     this.state = {
       showToolTip: false
     };
     this.showToolTip = this.showToolTip.bind(this);
     this.hideToolTip = this.hideToolTip.bind(this);
-    this.setSource = this.setSource.bind(this);
-    this.setText = this.setText.bind(this);
-    this.setSource(this.props.icon);
-    this.setText(this.props.icon);
+    this.getSource = this.getSource.bind(this);
+    this.getText = this.getText.bind(this);
   }
 
   public render(): JSX.Element {
-    const getIconColour = () => {
+    const iconColor = (() => {
       if(this.props.isSet) {
         return 'purple';
       }
       return 'grey';
-    };
+    })();
+    const source = (() => {
```

Comment:
If you're going to assign these to variables then do it directly instead of construction a function expression.

```const source = this.whatever;```

However in this case you can inline the expression directly into the attribute.